### PR TITLE
ENH: stats: add `axis` tuple and `nan_policy` to `kurtosis`

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
#### Reference issue

gh-14651

#### What does this implement/fix?

Adds `_axis_nan_policy_factory` to `kurtosis`. `kurtosis` tests looked quite strong so I didn't add any special tests.

### CODE REVIEW

**Review by mdhaber — APPROVED** [review]
Even though `kurtosis` already has `nan_policy` and `axis`, I see several benefits of adding the decorator. 

Previously, `kurtosis` would return a masked array when the input was a regular array with NaNs and `nan_policy='omit'`. E.g.,
```python3
np.random.seed(0)
random_array = np.random.rand(4, 6)
random_mask = np.random.rand(4, 6) > 0.8
random_nan_array = random_array.copy()
random_nan_array[random_mask] = np.nan
stats.kurtosis(random_nan_array, nan_policy='omit', _no_deco=True)
```
prints
```
masked_array(data=[-0.93590102, -0.80903849, -1.5       , -1.5       ,
                   -1.00060498, -1.49224995],
             mask=False,
       fill_value=1e+20)
```

Now, 
```
>>> stats.kurtosis(random_nan_array, nan_policy='omit') 
array([-0.93590102, -0.80903849, -1.5       , -1.5       , -1.00060498,
       -1.49224995])
```

Treating `nan_policy='omit'` with NaNs also lead to the appearance of the unusual 'masked constant' in the case of an empty slice.
```
>>> stats.kurtosis([np.nan, np.nan, np.nan], nan_policy='omit', _no_deco=True)
masked
```
Now:
```
>>> stats.kurtosis([np.nan, np.nan, np.nan], nan_policy='omit')
nan
```
as expected for an empty slice.

Axis tuples are now supported. 

Previously, masked arrays were accepted without warning, but the masked values were included in the calculation; now they are accepted and the calculation is performed correctly.

@tupui @Kai-Striega Care to play with this a bit? I figure at this point you probably have your own preferred battery of tests for this sort of thing.

<details>

```python3
import numpy as np
from scipy import stats
from numpy.ma import masked_array as ma

fun = stats.kurtosis

def try_case(x, message, kwds):
    try:
        print(message)
        res = fun(x, **kwds)
        print(res)
        print(type(res), res.dtype)
    except AttributeError:
        print(type(res))
    except Exception as e:
        print(f"{type(e)}: {str(e)}")

    try:
        print("vs (master)")
        res = fun(x, **kwds,_no_deco=True)
        print(res)
        print(type(res), res.dtype)
    except AttributeError:
        print(type(res))
    except Exception as e:
        print(f"{type(e)}: {str(e)}")

    print("----")

np.random.seed(0)
random_array = np.random.rand(4, 6)
random_mask = np.random.rand(4, 6) > 0.8
random_nan_array = random_array.copy()
random_nan_array[random_mask] = np.nan
random_ma = np.ma.masked_array(random_array, mask=random_mask)
random_ma32 = np.ma.masked_array(random_array, mask=random_mask, dtype=np.float32)

cases = [
          ([], 'Zero Observations:', {}),
          ([1.], 'One Observation:', {}),
          ([1., 2.], 'Two Observations', {}),
          (random_array, '2D Array, no axis', {}),
          (random_array, '2D Array, axis=0', {'axis': 0}),
          (random_array, '2D Array, axis=1', {'axis': 1}),
          (np.matrix(random_array), 'Matrix, axis=1', {'axis': 1}, ),
          (random_array, '2D Array, axis=None', {'axis': None}),
          (random_array, '2D Array, axis=(0,1)', {'axis': (0, 1)}),
          ([np.nan, 1., 2., 3.], '1D Array with NaN, No nan_policy', {}),
          ([np.nan, 1., 2., 3.], '1D Array with NaN, raise', {'nan_policy': 'raise'}),
          ([np.nan, 1., 2., 3.], '1D Array with NaN, propagate', {'nan_policy': 'propagate'}),
          ([np.nan, 1., 2., 3.], '1D Array with NaN, omit', {'nan_policy': 'omit'}),
          ([np.nan, np.nan, np.nan], '1D Array with All NaNs, omit', {'nan_policy': 'omit'}),
          ([np.nan, np.nan, 3.], '1D Array with all but one NaN, omit', {'nan_policy': 'omit'}),
          (ma([0., 1., 2., 3.], mask=[True, False, False, False]), '1D Masked array, one masked', {}),
          (ma([1., 2., 3.], mask=[True, True, True]), '1D Masked array, all masked', {}),
          (ma([1., 2., 3.], mask=[True, True, False]), '1D Masked array, all but one masked', {}),
          (random_nan_array, '2D Array with NaNs, propagate, axis=1', {'axis': 1, 'nan_policy': 'propagate'}, ),
          (random_nan_array, '2D Array with NaNs, omit, axis=1', {'axis': 1, 'nan_policy': 'omit'}, ),
          (random_ma, '2D Masked array, axis=1', {'axis': 1}, ),
          (random_array.astype(np.float32), '2D Array, float32, axis=1', {'axis': 1}),
          (random_nan_array.astype(np.float32), '2D Array with NaNs, float32, propagate, axis=1', {'axis': 1, 'nan_policy': 'propagate'}, ),
          (random_ma32, '2D Masked Array, float32, axis=1', {'axis': 1}, ),
          ]

for case in cases:
    try_case(*case)
```
</details>

File: scipy/stats/tests/test_axis_nan_policy.py [hunk 1]
```diff
@@ -26,7 +26,8 @@
     (stats.wilcoxon, tuple(), dict(), 1, 2, True, None),
     (stats.gmean, tuple(), dict(), 1, 1, False, lambda x: (x,)),
     (stats.hmean, tuple(), dict(), 1, 1, False, lambda x: (x,)),
-    ]
+    (stats.kurtosis, tuple(), dict(), 1, 1, False, lambda x: (x,))
```
Comment (Kai-Striega) [inline scipy/stats/tests/test_axis_nan_policy.py:29]: nitpicking here, but this should have a trailing comma. That way it won't show up on VCS as a change when additional stats functions are added.

**Review by Kai-Striega — APPROVED** [review]
Except for a nitpicky comment everything looks good to me.

@mdhaber test suite is actually more comprehensive that what I've been using (I'll be stealing some of his) the only additional test that I have is with `object` typed arrays but those pass too.

**Comment by mdhaber** [comment]
OK, I don't think it's worth CI to add that comma here, so next time. Thanks everyone!

**Pull request merged**